src/modules/persistent-store.ts

```diff
@@ -1,25 +1,37 @@
 import { Store } from 'store-unit';
 import * as browserStorage from 'src/background/webapis/storage';
 
+type Options<S> = {
+  retrieve: (key: string) => Promise<S | undefined>;
+  save: (key: string, value: S) => Promise<void>;
+};
+
 export class PersistentStore<T> extends Store<T> {
+  static async readSavedState<T>(key: string) {
+    return browserStorage.get<T>(key);
+  }
+
   protected key: string;
   protected isReady: boolean;
```

Comment:
It looks like we can make these fields private again

Reply:
Yep